pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2818,6 +2819,30 @@ func handlePhysicalIOAdapterListImpl(ctxArg interface{}, key string,
 					}
 					aa.AddOrUpdateIoBundle(log, vfIb)
 				}
+			} else if ib.Type == types.IoVCAN {
+				// Initialize (create and enable) Virtual CAN device
+				vcan, err := canbus.AddVCANLink(ib.Ifname)
+				if err != nil {
+					log.Errorf("Failed to add Virtual CAN interface: %s", err)
+				}
+				err = canbus.LinkSetUp(vcan)
+				if err != nil {
+					log.Errorf("Failed to enable virtual CAN interface: %s", err)
+				}
+			} else if ib.Type == types.IoCAN {
+				// Initialize physical CAN device
+				canIf, err := canbus.GetCANLink(ib.Ifname)
+				if err != nil {
+					log.Errorf("Failed to fetch CAN interface: %s", err)
+				}
+				err = canbus.SetupCAN(canIf, ib.Cbattr)
```

Comment:
```suggestion
				err = canbus.SetupCAN(canIf, ib.Cbattr)
				ib.Error = err
				ib.ErrorTime = time.Now()
```

this reports the error back to the controller and IMO makes it easier for them to find mistakes, like f.e. when they put in a float instead of a uint32.

Reply:
`ib.Error = err` doesn't work because err is of type error while ib.Error is a string. But I liked the suggestion, I will convert it and also set ib.Error for other cases as well (VCAN interfaces) and functions....

Reply:
Done.